pkg/scheduler/plugins/capacity/capacity.go

```diff
@@ -220,133 +434,339 @@ func (cp *capacityPlugin) OnSessionOpen(ssn *framework.Session) {
+		if childAttr.capability == nil {
+			childAttr.realCapability = realCapability
+		} else {
+			realCapability.MinDimensionResource(childAttr.capability, api.Infinity)
+			childAttr.realCapability = realCapability
 		}
-		ssn.RecordPodGroupEvent(job.PodGroup, v1.EventTypeNormal, string(scheduling.PodGroupUnschedulableType), "queue resource quota insufficient")
-		return util.Reject
-	})
+		oldDeserved := childAttr.deserved.Clone()
+		childAttr.deserved.MinDimensionResource(childAttr.realCapability, api.Infinity)
+		childAttr.deserved.MinDimensionResource(childAttr.request, api.Infinity)
 
-	// Register event handlers.
-	ssn.AddEventHandler(&framework.EventHandler{
-		AllocateFunc: func(event *framework.Event) {
-			job := ssn.Jobs[event.Task.Job]
-			attr := cp.queueOpts[job.Queue]
-			attr.allocated.Add(event.Task.Resreq)
-			metrics.UpdateQueueAllocated(attr.name, attr.allocated.MilliCPU, attr.allocated.Memory)
+		childAttr.deserved = helpers.Max(childAttr.deserved, childAttr.guarantee)
+		totalDeserved.Sub(oldDeserved).Add(childAttr.deserved)
+	}
 
-			cp.updateShare(attr)
+	// Check if the parent queue's deserved resources are less than the total deserved resources of child queues
+	if attr.deserved.LessPartly(totalDeserved, api.Zero) {
+		return fmt.Errorf("deserved resources of queue <%s> are less than the sum of its child queues' deserved resources", attr.name)
+	}
 
-			klog.V(4).Infof("Capacity AllocateFunc: task <%v/%v>, resreq <%v>,  share <%v>",
-				event.Task.Namespace, event.Task.Name, event.Task.Resreq, attr.share)
-		},
-		DeallocateFunc: func(event *framework.Event) {
-			job := ssn.Jobs[event.Task.Job]
-			attr := cp.queueOpts[job.Queue]
-			attr.allocated.Sub(event.Task.Resreq)
-			metrics.UpdateQueueAllocated(attr.name, attr.allocated.MilliCPU, attr.allocated.Memory)
+	// Check if the parent queue's guarantee resources are less than the total guarantee resources of child queues
+	if attr.guarantee.LessPartly(totalGuarantee, api.Zero) {
+		return fmt.Errorf("guarantee resources of queue <%s> are less than the sum of its child queues' guarantee resources", attr.name)
+	}
 
-			cp.updateShare(attr)
+	for _, childAttr := range attr.children {
+		err := cp.checkHierarchicalQueue(childAttr)
+		if err != nil {
+			return err
+		}
+	}
 
-			klog.V(4).Infof("Capacity EvictFunc: task <%v/%v>, resreq <%v>,  share <%v>",
-				event.Task.Namespace, event.Task.Name, event.Task.Resreq, attr.share)
-		},
+	return nil
+}
+
+func (cp *capacityPlugin) createOrUpdateRootQueue(vcClient vcclient.Interface) error {
+	queue, err := vcClient.SchedulingV1beta1().Queues().Get(context.TODO(), cp.rootQueue, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			queue = &vcv1beta1.Queue{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: string(cp.rootQueue),
+				},
+				Spec: vcv1beta1.QueueSpec{},
+			}
+			cp.makeRootQueue(queue)
+
+			return retry.OnError(wait.Backoff{
+				Steps:    60,
```

Comment:
The default `defaultSchedulerPeriod` for `runOnce()` is 1 second. Is the number of `Steps` set for retries too large here?